public/app/features/dashboard-scene/scene/LibraryVizPanel.tsx

```diff
@@ -70,12 +80,36 @@ export class LibraryVizPanel extends SceneObjectBase<LibraryVizPanelState> {
           }),
           new PanelNotices(),
         ],
-      });
+      };
+
+      const panel = new VizPanel(vizPanelState);
+      const gridItem = this.parent;
+      if (libPanelModel.repeat && gridItem instanceof SceneGridItem && gridItem.parent instanceof SceneGridLayout) {
+        this._parent = undefined;
+        const repeater = new PanelRepeaterGridItem({
+          key: gridItem.state.key,
+          x: libPanelModel.gridPos.x,
+          y: libPanelModel.gridPos.y,
+          width: libPanelModel.repeatDirection === 'h' ? 24 : libPanelModel.gridPos.w,
+          height: libPanelModel.gridPos.h,
+          itemHeight: libPanelModel.gridPos.h,
+          source: this,
+          variableName: libPanelModel.repeat,
+          repeatedPanels: [],
+          repeatDirection: libPanelModel.repeatDirection === 'h' ? 'h' : 'v',
+          maxPerRow: libPanelModel.maxPerRow,
+        });
+        gridItem.parent.setState({
+          children: gridItem.parent.state.children.map((child) =>
+            child.state.key === gridItem.state.key ? repeater : child
+          ),
+        });
+      }
```

Comment:
@kaydelaney what is going on here? this does not make any sense. 

repeat option is not part of the library panel it's part of the dashboard definition it's in

Reply:
> repeat option is not part of the library panel it's part of the dashboard definition it's in

Is it? From my grafana.db:
<img width="1470" alt="image" src="https://github.com/grafana/grafana/assets/45561153/6602ea76-7cd6-4da2-b096-754a539cc6fc">

Reply:
ah, it's part of the library panel. that makes it messy I see :(

Reply:
@kaydelaney yea, sorry I did not know. I thought it was outside it so we could initialize a PanelRepeaterGridItem and repeat a LibraryVizPanel in an agnostic way (not caring if it's a VizPanel or LibraryVizPanel). 

but I see now that since the repeat option is part of the library panel definition and we only know that after LibraryVizPanel is activated this get's very messy